src/parallel/LoadCalc.cpp

```diff
@@ -226,6 +227,44 @@ arma::vec nnls(const arma::mat &A, const arma::vec &b, int max_iter = 500, doubl
 	}
 	return x;
 }
+
+arma::vec nnls_lawson_hanson(const arma::mat &A, const arma::vec &b) {
+	int m = static_cast<int>(A.n_rows);
+	int n = static_cast<int>(A.n_cols);
+	// A_arr has to be column-major. This is ensured by arma::vectorise, as it returns a column-major matrix by default.
+	// (arma stores matrices column-major)
+	auto A_arr = arma::conv_to<std::vector<double>>::from(arma::vectorise(A));
+	auto b_vec = arma::conv_to<std::vector<double>>::from(b);
+	std::vector<double> w_vec(n);
+	std::vector<double> zz_vec(m);
+	std::vector<int> index_vec(n);
+	std::vector<double> x_vec(n);
+	int mode;
+	double rnorm;
+	nnls_c(A_arr.data(), &m, &m, &n, b_vec.data(), x_vec.data(), &rnorm, w_vec.data(), zz_vec.data(), index_vec.data(),
+		   &mode);
+	std::cout << "lawson_hanson: mode " << mode << std::endl;
+	std::cout << "lawson_hanson: rnorm " << rnorm << std::endl;
+	return x_vec;
+}
+
+arma::vec nnls(const arma::mat &A, const arma::vec &b){
+	auto coordinate_wise_solution = nnls_coordinate_wise(A, b);
+	auto coordinate_wise_residual = A * coordinate_wise_solution - b;
+	auto coordinate_wise_residual_norm = arma::norm(coordinate_wise_residual);
+	std::cout << "coordinate_wise_residual_norm = " << coordinate_wise_residual_norm << std::endl;
+	auto lawson_hanson_solution = nnls_lawson_hanson(A, b);
+	auto lawson_hanson_residual = A * lawson_hanson_solution - b;
+	auto lawson_hanson_residual_norm = arma::norm(lawson_hanson_residual);
+	std::cout << "lawson_hanson_residual_norm   = " << lawson_hanson_residual_norm << std::endl;
+	if (coordinate_wise_residual_norm < lawson_hanson_residual_norm) {
+		std::cout << "returning coordinate_wise_solution" << std::endl;
+		return coordinate_wise_solution;
+	} else {
+		std::cout << "returning lawson_hanson_solution" << std::endl;
+		return lawson_hanson_solution;
+	}
+}
```

Comment:
does it really make sense to always compute both solutions?

Reply:
for test reasons: yes :)
If at some point we notice that one of those is always better, we'll change it.